### PR TITLE
JUP-24 - Event Description Page

### PR DESCRIPTION
**Pull request by BK2004**
- Added individual pages for events under /event/[eventId]
- Added byId procedure to events API router
- Event pages link to host club through titles

### CODE REVIEW

**Review by RubenOlano — APPROVED** [review]
Look good 👍🏽

File: src/pages/event/[eventId].tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    </>;
+};
+export default EventPage;
+
+import { api } from "@src/utils/api";
+import { db } from "@src/server/db";
+import { appRouter } from "@src/server/api/root";
+import { createInnerTRPCContext } from "@src/server/api/trpc";
+import { createServerSideHelpers } from "@trpc/react-query/server";
+import { type GetServerSidePropsContext, type GetStaticPaths, type InferGetStaticPropsType } from "next";
```
Comment (Master7130) [inline src/pages/event/[eventId].tsx:67]: @RubenOlano Do we want all the imports at the top of the page?

File: src/pages/event/[eventId].tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                            Contact Us
+                        </button>
+                        <button className="py-1 px-3 rounded-lg border-black border-[1px] bg-white">
+                            Register
+                        </button>
+                    </div>
+                </div>
+            </div>
+        </main>
+    </>;
```
Comment (Master7130) [inline src/pages/event/[eventId].tsx:58]: Parenthesis around the jsx would be nice.

File: src/pages/event/[eventId].tsx (hunk 1 above)
Comment (RubenOlano) [reply src/pages/event/[eventId].tsx:67]: It should be fine since all the imports there relate the static props stuff

**Comment by nl32** [comment]
same issue as with clubs where the static paths don't update for new events. @RubenOlano

**Comment by RubenOlano** [comment]
> same issue as with clubs where the static paths don't update for new events. @RubenOlano

I'm going to try pushing something to the branch. We don't really need trpc for this page so I'm gonna try just fetching it from the server and passing it through `getServerSideProps`

**Comment by RubenOlano** [comment]
Ok, I had to rework it a little bit but it now works with `getServerSIdeProps`. For future reference @BK2004 you can use `with` in drizzle queries to grab the object its referencing

File: src/pages/event/[eventId].tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        <div className="mb-5 flex flex-col space-y-6 px-7">
+          <div className="relative h-full w-full rounded-xl bg-slate-50 p-10 shadow-lg">
+            <Image src={club.image} width={100} height={100} alt="club image" />
+            <hr className="my-3 mt-5 h-[1px] border-0 bg-black" />
+            <div className="mb-3 flex w-full justify-between">
+              <div>
+                <a
+                  className="text-4xl font-semibold"
+                  href={`/directory/${club.id}`}
+                >
```
Comment (RubenOlano) [inline src/pages/event/[eventId].tsx:32]: I would change this to a NextJS Link element

**Review by RubenOlano — CHANGES_REQUESTED** [review]

**Review by BK2004 — COMMENTED** [review]
👍

**Review by RubenOlano — APPROVED** [review]
Good job 👍🏽

**Review by Master7130 — APPROVED** [review]

**Pull request merged**